lib/kube/proxy/remotecommand.go

```diff
@@ -107,11 +109,17 @@ func createRemoteCommandProxy(req remoteCommandRequest) (*remoteCommandProxy, er
 	if err != nil {
 		return nil, trace.Wrap(err)
 	}
+	defer proxy.Close()
+
 	if proxy.resizeStream != nil {
 		proxy.resizeQueue = newTermQueue(req.context, req.onResize)
 		go proxy.resizeQueue.handleResizeEvents(proxy.resizeStream)
 	}
-	return proxy, nil
+	err = exec(proxy)
+	if err := proxy.sendStatus(err); err != nil {
+		log.Warningf("Failed to send status: %v", err)
+	}
+	return nil, nil
```

Comment:
This is intentional. The `nil /* rsp */, nil /*error*/`must exist because otherwise the middleware will try to write the returned error or response into the hijacked responsewriter